modeling/inference_model.py

```diff
@@ -0,0 +1,601 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+import time
+from typing import List, Optional, Union
+from logger import logger
+
+import torch
+import numpy as np
+import transformers
+from transformers import (
+    GPT2Tokenizer,
+    AutoTokenizer,
+)
+
+import utils
+
+try:
+    import tpu_mtj_backend
+except ModuleNotFoundError as e:
+    # Not on TPU... hopefully
+    if utils.koboldai_vars.use_colab_tpu:
+        raise e
+
+# I don't really like this way of pointing to the current model but I can't
+# find a way around it in some areas.
+current_model = None
```

Comment:
would having `HFTorchInferenceModel` just access the tokenizer with `self.tokenizer` work?

Reply:
looks like it! i think this was a byproduct of originally not having the patch functions be methods of `HFTorchInferenceModel`, but now they are. good catch!